### PR TITLE
Checks for a list of sobjects and the corresponding create/edit/view qa directories.

### PR DESCRIPTION
**Pull request by dbreese**
May require some tweaking during integration, but this should be a good stepping stone.

@sfdctaka @khawkins @maliroteh-sf

### CODE REVIEW

**Comment by dbreese** [comment]
Hmmm, not sure why "templateChooserCommand" test is failing.

**Comment by dbreese** [comment]
Appears to possibly be a flapper. I just re-ran the failed test and it passed with no issues, so we can keep an eye on ie.

File: resources/instructions/createSObjectLwcQuickActions.html [hunk 1]
```diff
@@ -1,81 +1,110 @@
 <!doctype html>
 <html lang="en">
-    <head>
```
Comment (dbreese) [inline resources/instructions/createSObjectLwcQuickActions.html:3]: Sorry, most of the changes is just the addition of the scripts in the html. I made mistake of reformatting doc without realizing.

File: resources/instructions/createSObjectLwcQuickActions.html [hunk 2]
```diff
@@ -1,81 +1,110 @@
...
-                    'generateLwcQuickActionsButton'
-                );
-            });
-        </script>
-        <script src="--- MESSAGING_SCRIPT_SRC ---"></script>
-    </body>
-</html>
+        });
+
+        // Wait until all scripts are loaded, before engaging with e.g.
```
Comment (dbreese) [inline resources/instructions/createSObjectLwcQuickActions.html:76]: This is the meat of the change here -- makes a call to new function and dynamically generates table rows.

File: src/commands/wizard/lwcGenerationCommand.ts [hunk 3]
```diff
@@ -22,9 +35,81 @@ export class LwcGenerationCommand {
                             panel.dispose();
                             return resolve();
                         }
+                    },
+                    {
+                        type: 'getQuickActionStatus',
+                        action: async (_panel, data, callback) => {
+                            // TODO: Hook this up to function from W-14439417 that parses landing_page.json.
+                            const sobjects = [
```
Comment (dbreese) [inline src/commands/wizard/lwcGenerationCommand.ts:43]: Just a hard-coded list for now.

File: src/commands/wizard/lwcGenerationCommand.ts [hunk 4]
```diff
@@ -22,9 +35,81 @@ export class LwcGenerationCommand {
                             panel.dispose();
                             return resolve();
                         }
+                    },
+                    {
+                        type: 'getQuickActionStatus',
+                        action: async (_panel, data, callback) => {
```
Comment (khawkins) [inline src/commands/wizard/lwcGenerationCommand.ts:41]: Nit: `data` will probably go to `_data` too, unless you anticipate passing any in.

File: src/commands/wizard/lwcGenerationCommand.ts [hunk 5]
```diff
@@ -22,9 +35,81 @@ export class LwcGenerationCommand {
...
             );
         });
     }
+
+    static async checkForExistingQuickActions(
+        sobjects: string[]
+    ): Promise<SObjectQuickActionStatus> {
+        return new Promise<SObjectQuickActionStatus>(async (resolve) => {
+            const results: SObjectQuickActionStatus = { sobjects: {} };
+            results.sobjects = {};
```
Comment (khawkins) [inline src/commands/wizard/lwcGenerationCommand.ts:68]: You don't need this, right? looks like it's initialized as such in your declaration.

File: src/commands/wizard/lwcGenerationCommand.ts [hunk 6]
```diff
@@ -22,9 +35,81 @@ export class LwcGenerationCommand {
...
+
+            return resolve(results);
+        });
+    }
+
+    private static checkForExistingQuickAction(
+        sobject: string,
+        qaName: string
+    ): boolean {
+        const expectedDirName = `${sobject}.${qaName}.quickAction-meta.xml`;
```
Comment (khawkins) [inline src/commands/wizard/lwcGenerationCommand.ts:103]: Nit: this variable name could probably be slightly more appropriate, e.g. `expectedMetadataFilename` or somesuch.

**Review by khawkins — APPROVED** [review]
Approach looks good to me overall! Certainly in line with what we did for landing page template selection.

**Pull request merged**